libraries/standard/mqtt/src/mqtt.c

```diff
@@ -313,6 +313,247 @@ static MQTTStatus_t validatePublishParams( const MQTTContext_t * pContext,
                                            const MQTTPublishInfo_t * pPublishInfo,
                                            uint16_t packetId );
 
+/**
+ * @brief Handle special corner cases for wildcards at the end of topic
+ * filters, as documented by the MQTT protocol spec.
+ *
+ * It concludes a match between the topic name and topic filter for the
+ * following special cases:
+ * - When the topic filter ends with "/+" or "/#" characters, but the topic
+ * name only ends with '/'.
+ * - When the topic filter ends with "/#" characters, but the topic name
+ * ends at the parent level.
+ *
+ * @param[in] pTopicFilter The topic filter containing the wildcard.
+ * @param[in] topicFilterLength Length of the topic filter being examined.
+ * @param[in] filterIndex Index of the topic filter being examined.
+ * @param[in] topicNameLength Length of the topic name being examined.
+ * @param[in] nameIndex Index of the topic name being examined.
+ *
+ * @return Returns whether the topic filter and the topic name match.
+ */
+static bool matchWildcardsSpecialCases( const char * pTopicFilter,
```

Comment:
Iirc, in v4_beta wasn't this called `matchEndWildcards`? Why was it changed? Imo that was a more descriptive name since it immediately tells me that it checks for wildcards at the end of the filter.

Reply:
The `matchEndWildcards` can be misleading as it may suggest that it covers _all_ cases of topic filters ending with wildcards. This function only covers some special cases, whereas the  `matchWildcards` function covers the _nominal_ cases of wildcards  (which includes wildcards at the end of the topic filter)

Reply:
`matchWildcardsSpecialCases` gives less information as to what the function is doing that `matchEndWidcards`. The name could be improved imo. `matchEndWildcardsSpecialCases`?

Reply:
Sure, will update